tests/trainer/test_trainer.py

```diff
@@ -1596,20 +1596,52 @@ def test_pytorch_profiler_trainer_ddp(tmpdir, use_output_filename):
 
     profiler = PyTorchProfiler(output_filename=output_filename)
 
+    model = BoringModel()
+    trainer = Trainer(fast_dev_run=True, profiler=profiler, accelerator="ddp", gpus=2, logger=TensorBoardLogger(tmpdir))
+    trainer.fit(model)
+
+    enabled = use_output_filename or not use_output_filename and profiler.local_rank == 0
+
+    if enabled:
+        assert len(profiler.summary()) > 0
+        expected = {'validation_step', 'training_step_and_backward', 'training_step', 'backward'}
+        assert set(profiler.profiled_actions.keys()) == expected
+    else:
+        assert profiler.summary() is None
+        assert set(profiler.profiled_actions.keys()) == set()
+
+    if use_output_filename:
+        profiler.describe()
+        data = Path(profiler.output_fname).read_text()
+        assert len(data) > 0
+
+
+@pytest.mark.parametrize("use_output_filename", [True])
+def test_pytorch_profiler_trainer(tmpdir, use_output_filename):
+    """Ensure that the profiler can be given to the training and default step are properly recorded. """
+
+    if use_output_filename:
```

Comment:
You are passing only `True` in `parametrize`

Reply:
Done !